### PR TITLE
Refactored JavaSerializationHelper for https://github.com/pac4j/play-pac4j/issues/252

### PR DESCRIPTION
**Pull request by unkarjedy**
@leleuj
This is a first PR related to play-pack4j issue:
https://github.com/pac4j/play-pac4j/issues/252

### CODE REVIEW

File: pac4j-core/src/test/java/org/example/DummyValue.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package org.example;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * This class should be in a package different from org.pack4j.
```
Comment (leleuj) [inline pac4j-core/src/test/java/org/example/DummyValue.java]: `pac4j`

File: pac4j-core/src/test/java/org/example/DummyValue.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ * This is needed to test untrusted package deserialization feature
```
Comment (leleuj) [inline pac4j-core/src/test/java/org/example/DummyValue.java]: Please add the `@since` and `@author` annotations.

Comment (unkarjedy) [reply pac4j-core/src/test/java/org/example/DummyValue.java]: Just woke up from holidays...
Ok, will do that on the weekends.

File: pac4j-core/src/main/java/org/pac4j/core/util/JavaSerializationHelper.java [hunk 3]
```diff
@@ -57,40 +64,96 @@ public String serializeToBase64(final Serializable o) {
...
+    /**
+     * Deserialize a base64 String into a Java object.
      *
      * @param base64 the serialized object as a base64 String
-     * @return the unserialized Java object
+     * @return the deserialized Java object
      */
-    public Serializable unserializeFromBase64(final String base64) {
-        return unserializeFromBytes(Base64.getDecoder().decode(base64));
+    public Serializable deserializeFromBase64(final String base64) {
```
Comment (leleuj) [inline pac4j-core/src/main/java/org/pac4j/core/util/JavaSerializationHelper.java]: I also realize we can't change the name in 3.5. Its breaks backward compatibility...

**Comment by leleuj** [comment]
Replaced by: https://github.com/pac4j/pac4j/pull/1243 and https://github.com/pac4j/pac4j/pull/1244

**Pull request closed without merging**